spec/haskell/src/SEL4/Object/Reply.lhs

```diff
@@ -95,14 +95,13 @@ This module specifies the behavior of reply objects.
 >     when (nextReplyPtrOpt /= Nothing) $ do
 >         assert (isHead nextReplyPtrOpt) "the reply must be at the head"
 >         scPtr <- return $ theHeadScPtr nextReplyPtrOpt
->         tcbScPtrOpt <- threadGet tcbSchedContext tcbPtr
->         when (tcbScPtrOpt == Nothing) $ schedContextDonate scPtr tcbPtr
 >         sc <- getSchedContext scPtr
 >         setSchedContext scPtr (sc { scReply = prevReplyPtrOpt })
 >         when (prevReplyPtrOpt /= Nothing) $ do
 >             prevReplyPtr <- return $ fromJust prevReplyPtrOpt
->             prevReply <- getReply prevReplyPtr
->             setReply prevReplyPtr (prevReply { replyNext = replyNext reply })
+>             updateReply prevReplyPtr (\reply -> reply { replyNext = replyNext reply })
```

Comment:
Update the lambda naming to avoid conflict:
`>             updateReply prevReplyPtr (\replyPrev -> replyPrev { replyNext = replyNext reply })`

Reply:
ah, yes, that’s is correct. I’ll fix it up.